test/HydraChain/ValidatrosData.test.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable node/no-extraneous-import */
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import { expect } from "chai";
+
+import { ERRORS } from "../constants";
+
+export function RunValidatorsDataTests(): void {
+  it("should revert syncValidatorsData if not called by system", async function () {
+    const { hydraChain } = await loadFixture(this.fixtures.initializedHydraChainStateFixture);
+    await expect(hydraChain.connect(this.signers.governance).syncValidatorsData([]))
+      .to.be.revertedWithCustomError(hydraChain, ERRORS.unauthorized.name)
+      .withArgs(ERRORS.unauthorized.systemCallArg);
+  });
+
+  it("should syncValidatorsData and update validators data", async function () {
```

Comment:
Test all possible cases that can be entered. (e.g. update of the validators power, increase, decrease, etc.)